tests/trilinos_tpetra/direct_solver_01.cc

```diff
@@ -0,0 +1,460 @@
+  temp_solution = 0;
+  SolverControl solver_control(1000, 1e-12);
+  SolverCG<LinearAlgebra::TpetraWrappers::Vector<double>> solver(
+    solver_control);
+
+  solver.solve(system_matrix,
+               temp_solution,
+               system_rhs,
+               PreconditionIdentity());
+
+  constraints.distribute(temp_solution);
+  solution = temp_solution;
+
+  LinearAlgebra::TpetraWrappers::Vector<double> output(temp_solution);
+
+  // do CG solve for new rhs
+  temp_solution = 0;
+  solution      = 0;
+  solver.solve(system_matrix,
+               temp_solution,
+               system_rhs_two,
+               PreconditionIdentity());
+
+  constraints.distribute(temp_solution);
+  solution = temp_solution;
+
+  LinearAlgebra::TpetraWrappers::Vector<double> output_two(temp_solution);
+
+  // factorize matrix for direct solver
+  temp_solution = 0;
+  solution      = 0;
+
+  {
+    deallog.push("DirectKLU2");
+    LinearAlgebra::TpetraWrappers::SolverDirect<double>::AdditionalData data(
+      "KLU2", true);
+    LinearAlgebra::TpetraWrappers::SolverDirect<double> direct_solver(
+      solver_control, data);
+
+    Teuchos::ParameterList amesos2_params("Amesos2");
+    Teuchos::ParameterList klu2_params = amesos2_params.sublist("KLU2");
+    klu2_params.set("ColPerm", "NATURAL");
+
+    direct_solver.set_pararameter_list(amesos2_params);
+
+    direct_solver.initialize(system_matrix);
+
+    // do solve 1
+    direct_solver.solve(temp_solution, system_rhs);
+
+    constraints.distribute(temp_solution);
+    solution = temp_solution;
+
+    // calculate l2 errors
+    output.add(-1.0, temp_solution);
+
+    const double local_error = output.l2_norm();
+    const double global_error =
+      std::sqrt(Utilities::MPI::sum(local_error * local_error, MPI_COMM_WORLD));
+
+    deallog << "Norm of error in direct solve 1: " << global_error << std::endl;
+
+    // do solve 2 without refactorizing
+    temp_solution = 0;
+    solution      = 0;
+    direct_solver.solve(temp_solution, system_rhs_two);
+
+    constraints.distribute(temp_solution);
+    solution = temp_solution;
+
+    // calculate l2 errors
+    output_two.add(-1.0, temp_solution);
+
+    const double local_error_two  = output_two.l2_norm();
+    const double global_error_two = std::sqrt(
+      Utilities::MPI::sum(local_error_two * local_error_two, MPI_COMM_WORLD));
+
+    deallog << "Norm of error in direct solve 2: " << global_error_two
+            << std::endl;
+
+    deallog.pop();
+  }
+  {
+    deallog.push("DirectMEASLES");
+    LinearAlgebra::TpetraWrappers::SolverDirect<double>::AdditionalData data(
+      "MEASLES", true);
+    try
+      {
+        LinearAlgebra::TpetraWrappers::SolverDirect<double> direct_solver(
+          solver_control, data);
+      }
+    catch (dealii::ExceptionBase &exc)
+      {
+        deallog << "Error: " << std::endl;
+        exc.print_info(deallog.get_file_stream());
+      }
+  }
```

Comment:
This is purposefully a non-existent solver, right? Perhaps add a comment in front of it?

Reply:
Done.